### PR TITLE
octopus: Add etsf-io variant

### PR DESCRIPTION
**Pull request by iamashwin99**
etsf-io is an optional dependency of octopus.
This MR adds etsf-io as a variant to the octopus spack package.
A variant of this MR has been tested at https://github.com/fangohr/octopus-in-spack/pull/76 to check that octopus does pick up the etsf-io variant.

### CODE REVIEW

**Review by fangohr — APPROVED** [review]
Looks good to me.

**Review by bernhardkaindl — APPROVED** [review]
Approving after approval by @fangohr, Built on Ubuntu 20.04 with gcc-10.3.0

**Pull request merged**
